pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/lineage/LineageManager.java

```diff
@@ -70,4 +71,14 @@ void updateLineageForRevertReplaceSegments(TableConfig tableConfig, String linea
    */
   void updateLineageForRetention(TableConfig tableConfig, SegmentLineage lineage, List<String> allSegments,
       List<String> segmentsToDelete, Set<String> consumingSegments);
+
+  /**
+   * Get segment zk metadata groups for MergeRollupTask
+   * @param tableConfig
+   * @param lineage
+   * @param segments
+   * @return segment zk metadata groups
```

Comment:
Can you elaborate on the `segment zk metadata groups`?

Reply:
This basically returns multiple Group/List of SegmentZkMetadata. Each group/list have the segments that should be merged together, segments from different groups should not be merged.